Sources/MapboxCoreNavigation/ReplayLocationManager.swift

```diff
@@ -47,6 +47,17 @@ open class ReplayLocationManager: NavigationLocationManager {
     var startDate: Date?
     
     private var synthesizedLocation: CLLocation?
+
+    /**
+     A handler that is called when `ReplayLocationManager` finished replaying `locations` and about to start from the
+     beginning. Return false to stop replaying `locations`.
+     */
+    public var onReplayLoopCompleted: ((ReplayLocationManager) -> Bool)?
+
+    /**
+     A handler that is called on each replayed location along with the location index in `locations` array.
+     */
+    public var onTick: ((_ index: Int, CLLocation) -> Void)?
```

Comment:
What would be the use-cases for these? For example why `onReplayLoopCompleted` is a callback instead of a flag? What would `onTick` be used for what cannot be covered by `delegate.locationManager(_, didUpdateLocations:)`?

Reply:
This is mainly for tests. I made this vars internal at first and then thought it can be useful for devs. 

The use case is to replay locations with ReplayLocationManager. The delegate for `ReplayLocationManager` is `RouteController` and tests needs a way to get callbacks when replay is finished to continue assertions (`onReplayLoopCompleted` callback) and perform some logic on each location updates (`onTick` callback). 

wdyt?

Reply:
This will require having a lot of boilerplate in tests. I would rather make it internal for now and if a request comes, reconsider decision.